autopush/src/client.rs

```diff
@@ -356,7 +358,11 @@ where
                     uaid.and_then(|uaid| Uuid::parse_str(uaid.as_str()).ok()),
                     Broadcast::from_hashmap(broadcasts.unwrap_or_default()),
                 ),
-                _ => return Err("Invalid message, must be hello".into()),
+                _ => {
+                    return Err(Error::BroadcastError(
```

Comment:
I think this is the only Error I'd call a "`BroadcastError`" in this module, the others here are more general client errors. Maybe just make those `GeneralError`s?